contracts/hts-precompile/AtomicHTS.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import "./IHRC.sol";
+import "./HederaTokenService.sol";
+import "./IHederaTokenService.sol";
+
+/**
+ * @dev This contract contains multiple examples highlighting the utilization of the 
+ * HIP-551: Batch transactions using HTS calls via HederaTokenService Precompile contract.
+ * The batches rolls multiple HTS calls in one transaction that passes ACID test 
+ * (atomicity, consistency, isolation, and durability).
+ * Read more about HIP-551 at https://hips.hedera.com/hip/hip-551
+ */
+contract AtomicHTS is HederaTokenService {
+    /// events ///
+    event BatchAlpha(int associateResponseCode, int grantKYCResponseCode, int transferTokenResponseCode);
+    event BatchBeta(int transferTokenResponseCode, int approveResponseCode, int associateResponseCode, int grantKYCResponseCode, int transferFromResponseCode);
+    event BatchCharlie(int unfreezeTokenResponseCode, int grantKYCResponseCode, int transferTokenResponseCode, int freezeTokenResponseCode);
+    event BatchDelta(int wipeTokenResponseCode, int mintTokenResponseCode, int transferTokenResponseCode);
+    event BatchEcho(int mintTokenResponseCode, int unfreezeTokenResponseCode, int grantKYCResponseCode, int freezeTokenResponseCode);
+    event BatchFoxtrot(int associateResponseCode, int mintTokenResponseCode, int grantKYCResponseCode, int transferTokenResponseCode);
+
+
+    /**
+     * @dev associates, grant token KYC, and send an amount of fungible token to a receiver.
+     * - associateToken() -> grantTokenKYC() -> transferToken()
+     */
+    function batchAlpha(address token, address sender, address receiver, int64 amount) external {
+        (int associateResponseCode) = HederaTokenService.associateToken(receiver, token);
+        require(associateResponseCode == HederaResponseCodes.SUCCESS, "Failed to associate token.");
```

Comment:
Allow for already associated response code also

Reply:
Hmm sorry not quite clear on this. The .associateToken() only returns one response code which is expected to be 22 after associate the token to the account. How is the `TOKEN_ALREADY_ASSOCIATED_TO_ACCOUNT = 194` response code supposed to fit into this scenario?

Reply:
Won't [associatedToken(receiver, token)](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/IHederaTokenService.sol#L373) return `TOKEN_ALREADY_ASSOCIATED_TO_ACCOUNT` is the contract was already associated?
If so i'm saying instead of failing the logic should handle `SUCCESS` and `TOKEN_ALREADY_ASSOCIATED_TO_ACCOUNT` since the goal is to make sure the contract is associated. Subsequents runs of the same function should fail if done in this way

Reply:
Ahh I see what you are saying! Gotcha pushing the fix soon!